### ISSUE TITLE
Tuning problems on PatchboxOS

### ISSUE THREAD

**Issue by macbutch** [TheTechnobear/Mi4Pd#15]
Hello Mark,

Thanks for the amazing work you've done here. I am having some trouble with these modules on Patchbox OS with the Pisound. I've posted on the Blokas forums (https://community.blokas.io/t/orac-question-braids-out-of-tune/2294/3) but this might be a better place to ask - happy to do my own builds etc (but need a bit of help if you can spare it). My problem is that the modules are out of tune. Now, I've seen that they need a 48k samplerate and I've done that (and tried different options) and as far as I can tell I *am* running at 48k but the MI synths in Orac around about 130 cents sharp. So right now I've got them tuned down one semitone and I've tuned my other synths about ~30 cents or so. 

I'm running the Patchbox OS with the latest updates available. I'm not sure if this includes your latest patch with the samplerate checks or not. I guess, to be honest, I'm finding it hard to get started with diagnosis.

If you had a minute to point me in the right direction I'd appreciate it very much but I'm fully aware you have other demands on your time.

**Comment by TheTechnobear** [comment]
Latest commit should ‘fix’ this within the constraints of the MI code , and not resampling.  

You can just build this repo with cmake on a rPI and then copy the created shared libs over those installed on the pisound.

**Comment by macbutch** [comment]
Hi @TheTechnobear,

Thanks for your help and apologies for how long it's taken to get back to you. I've pulled down the latest code from this repository, recompiled (I had to add `-fPIC` to the build) and copied the files in `release/lib` over the existing files in `/usr/local/orac/externals` and `/usr/local/MEC`. 

That alone didn't make any difference that I can detect: it is still a bit more than 130 cents sharp.

I've spent a bit of time hacking a way at this - really trial and error at this stage - and just hardcoding the pitch offset to -176 (instead of +193) gets me really close to being in tune. I would love to better understand where these numbers come from. The code comments say that braids (and the others) run at 96k samplerate internally and Jack is configured for 96k so that suggests I shouldn't need a pitch offset at all but I definitely do! 

Here are the (useful) values I tested today:

| **`pitch_offset` Value** | **Result** |
| --- | --- |
| 0 | > 130 cents sharp |
| 193 | ~3 semitones sharp |
| -176 | *roughly* in tune |

I might actually be going mad here. If any of this happens to make sense to you please let me know. I'll do more testing with a different reference synth tomorrow and report back.

**Comment by macbutch** [comment]
Just following up to note that the tuning details are confirmed now with reference to a handful of synths now (Volca Bass, Digitakt + Digitone). I'ts probably not spot on but it's definitely close.

**Comment by TheTechnobear** [comment]
last time, I tested this on 48k and 44.1k, both work fine - I tested with a digital tuner.
so Im not sure what's going on. in your case.... I'd need to retest to determine exactly.


my last fix, basically checks for 48k (it really should check for multiples of 48k, e.g. 48/96) , in my test, these did NOT require an offset... whereas 44.1k did (... no other sample rates are supported/tested) 

the reason behind all this is the MI code has hard coded resource tables that assume 96k , so it appears 48k being exactly half this are 'roughly' ok.
now this is NEVER going to precise going down this 'hack' route, as I put in the code its an 'approximation' at best. but I found it was usable.

Ive one idea about why you are getting an issue....
It may be in dev that I have updated Orac... in particular the brds module.
at some point in time, I did these pitch offsets in the pure data patch, 
but then I reviewed this and realised it would be better to do this within this PD external.

so I suspect part of the issue you are getting is perhaps your version of the brds module still has this old pitch correction code in it.. 
this causes two problems
a) pitch correction is being done in two places
b) the pitch correction in the PD is fixed - I believe to the 44.1 k value iirc.

so I believe if you update Orac to the dev version it will likely resolve your issues.
